### PR TITLE
exposing topologySpreadConstraints on the helm chart

### LINKED ISSUE #3891: Expose the ability to set topologyspreadConstraints on the Deployment
**Is your feature request related to a problem? Please describe.**
I want to spread my pods out over AZs

**Describe the solution you'd like**
I'll expose this via the helm chart

### PR DESCRIPTION
**Pull request by bjoernw**
This allows users to set `topologySpreadConstraints` which can be used to more evenly spread pods out over nodes/AZs/etc. and prevent uneven utilization of federation pods.

Fixes #3891

---

**Checklist**

Complete the checklist (and note appropriate exceptions) before the PR is marked ready-for-review.

- [X] Changes are compatible[^1]
- [X] Documentation[^2] completed
- [X] Performance impact assessed and acceptable
- Tests added and passing[^3]
    - [ ] Unit Tests
    - [ ] Integration Tests
    - [X] Manual Tests

**Exceptions**
Tested helm chart template generation locally

*Note any exceptions here*
N/A
**Notes**
N/A

[^1]: It may be appropriate to bring upcoming changes to the attention of other (impacted) groups. Please endeavour to do this before seeking PR approval. The mechanism for doing this will vary considerably, so use your judgement as to how and when to do this.
[^2]: Configuration is an important part of many changes. Where applicable please try to document configuration examples.
[^3]: Tick whichever testing boxes are applicable. If you are adding Manual Tests, please document the manual testing (extensively) in the Exceptions.

### CODE REVIEW

**Comment by Meemaw** [comment]
Hey @garypen any blockers to get this in? Its a relatively small change, but super useful one!

**Comment by garypen** [comment]
> Hey @garypen any blockers to get this in? Its a relatively small change, but super useful one!

Two things:
 - I'm on vacation (back on Monday)
 - I was waiting to re-enable `kubeconform` (which I think may have just happened)

Expect something to happen next week.

File: helm/chart/router/values.yaml [hunk 1]
```diff
@@ -232,3 +232,12 @@ probes:
...
     initialDelaySeconds: 0
+
+topologySpreadConstraints: []
+# topologySpreadConstraints:
+#  - maxSkew: 1
+#    topologyKey: kubernetes.io/hostname
+#    whenUnsatisfiable: ScheduleAnyway
+#  - maxSkew: 1
+#    topologyKey: topology.kubernetes.io/zone
+#    whenUnsatisfiable: ScheduleAnyway
```
Comment (garypen) [inline helm/chart/router/values.yaml:243]:
```suggestion
# -- Sets the [topology spread constraints](https://kubernetes.io/docs/concepts/scheduling-eviction/topology-spread-constraints/) for Deployment pods
topologySpreadConstraints: []
```

**Review by garypen — CHANGES_REQUESTED** [review]
Looks good. If you don't mind making those changes we can commit this.

**Comment by bjoernw** [comment]
@garypen done

**Review by garypen — APPROVED** [review]

**Pull request merged**